python/tvm/relay/tensorrt.py

```diff
@@ -439,25 +445,47 @@ def reshape_whitelist_fn(attrs, args):  # pylint: disable=unused-variable
             print("reshape: new shape dims must be explicit.")
             return False
         if use_implicit_batch:
-            shape = list(map(int, args[0].checked_type.shape))
-            new_shape = list(map(int, attrs.newshape))
+            shape = args[0].checked_type.shape
+            new_shape = attrs.newshape
             if len(new_shape) == 0 or len(shape) == 0:
                 print("reshape: Can't reshape to or from scalar.")
                 return False
-            # TRT cannot modify batch dimension.
-            original_volume = np.prod(shape)
-            # First, resolve 0.
-            for i, value in enumerate(new_shape):
-                if value == 0:
-                    new_shape[i] = shape[i]
-            # Resolve -1.
-            for i, value in enumerate(new_shape):
-                if value == -1:
-                    new_shape[i] = original_volume // np.prod([x for x in new_shape if x != -1])
-            # Remove batch dimension and see if volumes match
-            if shape[0] != new_shape[0]:
-                print("reshape: can't modify batch dimension.")
-                return False
+
+            dynamic_reshape = False
+            for value in shape:
+                if isinstance(value, tvm.tir.expr.Any):
+                    dynamic_reshape = True
+                    break
+
+            if dynamic_reshape:
+                # Make sure that the batch dim is unmodified.
+                if int(new_shape[0]) < 0:
+                    for shape_val, new_shape_val in enumerate(shape[1:], new_shape[1:]):
+                        if not (
+                            isinstance(shape_val, int)
+                            and isinstance(new_shape_val, int)
+                            and int(shape_val) == int(new_shape_val)
+                        ):
+                            return False
+                return True
```

Comment:
If new_shape[0] is > 0, we still need to check it as best as we can.

Reply:
Done..

Reply:
Thanks, looks good